src/server/DedicatedServer.cpp

```diff
@@ -187,63 +200,29 @@ void DedicatedServer::processPackets()
 						// player iterator is invalid after erase
 					}
 					mMatchMaker.removePlayer(packet->playerId);
-				}
-				else
-				{
+					syslog(LOG_DEBUG, "Player removed (%d) from %s (%s), %d players available",
+						   packet_id, packet->playerId.toString().c_str(),
+						   player->second->getName().c_str(), (int)mPlayerMap.size());
 				}
 
-				int pid = packet->data[0];
-				syslog(LOG_DEBUG, "Connection %s closed via %d, %d clients connected now", packet->playerId.toString().c_str(),  pid, mConnectedClients);
+				syslog(LOG_DEBUG, "Connection closed (%d) from %s, %d clients connected now",
+					   packet_id, packet->playerId.toString().c_str(), getConnectedClients());
 				break;
 			}
-			// player connects to server
 			case ID_ENTER_SERVER:
-			{
-				RakNet::BitStream stream(packet->data, packet->length, false);
-
-				stream.IgnoreBytes(1);	//ID_ENTER_SERVER
-
-				auto newplayer = std::make_shared<NetworkPlayer>(packet->playerId, stream);
-
-				// add to player map. protect with mutex
-				{
-					std::lock_guard<std::mutex> lock( mPlayerMapMutex );
-					mPlayerMap[packet->playerId] = newplayer;
-				}
-				mMatchMaker.addPlayer(packet->playerId, newplayer);
-				syslog(LOG_DEBUG, "New player \"%s\" connected from %s ", newplayer->getName().c_str(), packet->playerId.toString().c_str());
-
-				// if this is a locally hosted server, any player that connects automatically joins an
-				// open game
-				if (mPlayerHosted && 
-				    mMatchMaker.getOpenGamesCount() != 0)
-				{
-					RakNet::BitStream stream;
-					stream.Write((unsigned char)ID_LOBBY);
-					stream.Write((unsigned char)LobbyPacketType::JOIN_GAME);
-					auto writer = createGenericWriter(&stream);
-					writer->uint32(mMatchMaker.getOpenGameIDs().front());
-					writer->generic<std::string>("");
-					mMatchMaker.receiveLobbyPacket( packet->playerId, stream );
-				}
-				break;
-			}
 			case ID_LOBBY:
-			{
-				/// \todo assert that the player send an ID_ENTER_SERVER before
-
-				// which player is wanted as opponent
-				RakNet::BitStream stream(packet->data, packet->length, false);
-				mMatchMaker.receiveLobbyPacket( packet->playerId, stream );
-				break;
-			}
 			case ID_BLOBBY_SERVER_PRESENT:
-			{
-				processBlobbyServerPresent( packet );
-				break;
-			}
+				try {
+					RakNet::BitStream stream(packet->data, packet->length, false);
+					stream.IgnoreBytes(1);	// ignore the message id
+					processSingleMessage(static_cast<MessageType>(packet_id), packet->playerId, stream);
+				} catch (const std::exception& error) {
+					syslog(LOG_ERR, "An error occurred while processing packet (%d) from %s: %s",
+						   packet_id, packet->playerId.toString().c_str(), error.what());
+				}
```

Comment:
Missing break

Reply:
good catch

Reply:
one question is still, should we disconnect the player that caused the exception in the first case. To make sure that this player's connection is not in some weird, undefined state.

Reply:
I'm not sure. In future cases this can be an overreaction.